query_ranges_to_vnodes_generator: optimize comparisons

bhalevy opened the pull request:
There is no need to test _it == _ranges.end()
every iteration, as this can be tested once to
reduce `n`, if needed.

Also, we already check for ranges.size() in
`process_one_range`, so that result can be
propagated back to the caller that can test it instead of checking the results vector size yet again.

Optimization, no backport required.

scylladb-promoter commented:
#### :red_circle: CI State: ABORTED

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9961/)
:white_check_mark: - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9961/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9961/)
#### Failed Tests (553/25478):
- [test_scan_paging_missing_limit.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9961/testReport/junit/%28root%29/test_scan/test_scan_paging_missing_limit_debug_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_scan_paging_missing_limit.debug.1)
- [test_scan_paging_missing_limit.debug.2](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9961/testReport/junit/%28root%29/test_scan/test_scan_paging_missing_limit_debug_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_scan_paging_missing_limit.debug.2)
- [test_scan_paging_missing_limit.dev.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9961/testReport/junit/%28root%29/test_scan/test_scan_paging_missing_limit_dev_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_scan_paging_missing_limit.dev.1)
- [test_scan_paging_missing_limit.dev.3](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9961/testReport/junit/%28root%29/test_scan/test_scan_paging_missing_limit_dev_3) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_scan_paging_missing_limit.dev.3)
- [test_scan_paging_missing_limit.release.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9961/testReport/junit/%28root%29/test_scan/test_scan_paging_missing_limit_release_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_scan_paging_missing_limit.release.1)
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9961/testReport/junit/boost.aggregate_fcts_test.test_aggregate_avg.1.xunit/xml/_empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9961/testReport/junit/boost.aggregate_fcts_test.test_aggregate_avg.1.xunit/xml/_empty__2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9961/testReport/junit/boost.aggregate_fcts_test.test_aggregate_avg.1.xunit/xml/_empty__3) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9961/testReport/junit/boost.aggregate_fcts_test.test_aggregate_avg.2.xunit/xml/_empty_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
- [[empty]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/9961/testReport/junit/boost.aggregate_fcts_test.test_aggregate_avg.2.xunit/xml/_empty__2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+[empty])
#### Build Details:
- Duration: 10 hr
- Builder: spider2.cloudius-systems.com

Annamikhlin commented:
ci run was stack - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/9961/ - I aborted it 
in addition, there are 553 failure

bhalevy commented:
In v2 ([5dce914](https://github.com/scylladb/scylladb/commit/5dce914ea0df5e61aa3f66f4f01b26a27368834c))
- rebased
- cleaned up
- query_ranges_to_vnodes_generator: call process_one_range only when there are ranges to return
  - prevents assertion failure that was hit with v1

bhalevy commented:
retracting this PR.  this isn't called on the hot path and therefore not worth the complication
